unix/x0vncserver/XDesktop.cxx

```diff
@@ -701,6 +711,9 @@ bool XDesktop::handleGlobalEvent(XEvent* ev) {
     if (cev->subtype != XFixesDisplayCursorNotify)
       return false;
 
+    if (!isCursorVisibleOnScreen)
+      return false;
```

Comment:
You seem to check this actively in the other places. Why not here?

Reply:
I guess you mean why I don't check it in `EnterNotfiy` and `LeaveNotify` events? It's because we want to either hide or show the cursor.

Reply:
No, I mean here. Why rely on the cached `isCursorVisibleOnScreen` here rather than calling `XQueryPointer()` and checking the current state directly?

It would mean not having to have that cached value at all.

Reply:
I was told that XQueryPointer() is an expensive call so that's why it is cached and not called again.

Reply:
I do it only once in the `Start()` as we might also show the cursor on the wrong screen initially.

Reply:
That bridge is long passed. You are now calling it in the most expensive place (each loop), but caching for the least expensive place (when the cursor image changes). :)

Reply:
It is called once per loop, so we can track where the cursor is on screen anyway. So, we are constantly spamming that call.

Reply:
I changed it to not use cached value.